sentry_sdk/client.py

```diff
@@ -454,12 +454,18 @@ def _should_capture(
     def _should_sample_error(
         self,
         event,  # type: Event
+        hint,  # type: Hint
     ):
         # type: (...) -> bool
-        not_in_sample_rate = (
-            self.options["sample_rate"] < 1.0
-            and random.random() >= self.options["sample_rate"]
-        )
+        sampler = self.options.get("events_sampler", None)
+
+        if callable(sampler):
+            with capture_internal_exceptions():
+                sample_rate = sampler(event, hint)
```

Comment:
If `sampler` here returns something that can't be compared to the `1.0` on line 468, we will throw a `TypeError`. Would be good to add a check that the type of `sample_rate` here is correct (`int`, `float`, `bool`, I assume) and log some kind of actionable message if that's not the case.

Reply:
@sentrivana if the `sampler` returns an invalid value, should we sample the event? I think it would make the most sense to sample the event, since the default behavior is to send all events to Sentry

Reply:
@szokeasaurusrex I agree 👍🏻